library/std/src/thread/local.rs

```diff
@@ -409,9 +424,17 @@ pub mod fast {
         }
 
         pub unsafe fn get<F: FnOnce() -> T>(&self, init: F) -> Option<&'static T> {
-            match self.inner.get() {
-                Some(val) => Some(val),
-                None => self.try_initialize(init),
+            // SAFETY: See the definitions of `LazyKeyInner::get` and
+            // `try_initialize` for more informations.
+            //
+            // The call to `get` is made safe because no mutable references are
+            // ever handed out and the `try_initialize` is dependant on the
+            // passed `init` function.
```

Comment:
> no mutable references are ever handed out

This is a requirement that must be handled by the caller, right? In that case you should write it in the comment.

Reply:
This behaves the same as the comments for lines 539/540, this is a different module used with another `cfg`.